### PR TITLE
WDPT BER Patrick

### PR DESCRIPTION
**Pull request by Patrick-Groot**

### CODE REVIEW

File: src/viking.js [hunk 1]
```diff
@@ -1,11 +1,87 @@
...
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
```
Comment (trebeljahr) [inline src/viking.js:24]: a ternary here might be nice :)

File: src/viking.js [hunk 2]
```diff
@@ -1,11 +1,87 @@
...
+    addViking(viking) {
+        this.vikingArmy.push(viking);
+    };
+    addSaxon(saxon) {
+        this.saxonArmy.push(saxon);
+    };
+    vikingAttack(){
+        const i = Math.floor(Math.random() * this.saxonArmy.length);
+        const randomSaxon = this.saxonArmy[i];
+        const j = Math.floor(Math.random() * this.vikingArmy.length);
```
Comment (trebeljahr) [inline src/viking.js:58]: you use this way of creating random indeces 4 times, time to refactor it into it's own function ^^

File: src/viking.js [hunk 3]
```diff
@@ -1,11 +1,87 @@
...
+        const randomViking = this.vikingArmy[j];
+        const endOfVikingAttack = randomSaxon.receiveDamage(randomViking.attack());
+        endOfVikingAttack;
```
Comment (trebeljahr) [inline src/viking.js:61]: line 61 does nothing. you can delete it.

File: src/viking.js [hunk 4]
```diff
@@ -1,11 +1,87 @@
...
+        if (randomViking.health <= 0) {
+            this.vikingArmy.splice(i,1);
+        }
+        return endOfSaxonAttack;
+    };
+    showStatus(){
+        if (this.saxonArmy.length === 0) {
+            return "Vikings have won the war of the century!"
+        }
+        else if (this.vikingArmy.length === 0) {
```
Comment (trebeljahr) [inline src/viking.js:83]: you can drop the elses here because you return from each of the if blocks anyways. 
this is called "early" return pattern.

**Review by trebeljahr — COMMENTED** [review]
good job :+1:

**Comment by Patrick-Groot** [comment]
> good job +1

Thank you for your detailed feedback on this Lab and the others! :)

**Pull request closed without merging**